manifests/config.pp

```diff
@@ -59,6 +59,38 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd  = "/usr/sbin/pam-config -q --mkhomedir | grep session:"
+      $pamconfig_check_cmd  = "/usr/sbin/pam-config -q --sss | grep session:"
+      $nsconfig_check_cmd = "/usr/bin/test $( /bin/grep -cE '(passwd|group|sudoers).*sss' /etc/nsswitch.conf ) -gt 2"
+      $nsconfig_cmd = "/bin/sed -i -e '/^\(passwd\|group\|sudoers\):/{/sss\b/ b;s/^\(.*\)$/\1 sss/}' /etc/nsswitch.conf"
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
+          path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+          unless      => $pamconfig_mkhomedir_check_cmd,
+        }
+
+      }
+
+      exec { 'pam-config -a --sss':
+        path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+        unless      => $pamconfig_check_cmd,
+      }
+
+      exec { 'echo "sudoers: file sss" >> /etc/nsswitch.conf':
```

Comment:
I'm not fond of this module touching nsswitch at all, but if we really wanted it, then I think it would be best to:
* Use the nsswitch module and NOT add it as a dependency in metadata.json, as some will not use it.
* Put an if around its use and make the default false
```puppet
class whatever (
  $manage_nsswitch = false,
){
...
  if $manage_nsswitch {
  # use nsswitch module to only add sssd
  }

...
}
``` 

Id check with @sgnl05 to see what he thinks... But I wouldn't dislike this too much...